src/cpp/transport/Server.cpp

```diff
@@ -152,7 +157,7 @@ template<typename EndPoint>
 bool Server<EndPoint>::enable_p2p(uint16_t p2p_port)
 {
     bool rv = false;
-    if (running_cond_)
+    if (has_p2p() && running_cond_)
```

Comment:
This check is not really necessary, as the Server class has a `return false` implementation for `init_discovery` and `init_p2p`